File: java/chainofresponsibility/RequestProcessor.java
package chainofresponsibility;

public class RequestProcessor {

    private final MiddlewareHandler middlewareChainHandler;

    public RequestProcessor(MiddlewareHandler middlewareChainHandler) {
        this.middlewareChainHandler = middlewareChainHandler;
    }

    public Response processRequest(Request request) {
        if (!middlewareChainHandler.handle(request)) {
            return new Response("Request failed!", false);
        }
        System.out.println("processing request");
        return new Response("Request Successfully Processed", true);
    }
}

File: PHP/Patterns/src/Behavioral/ChainOfResponsibility/RequestProcessor.php
<?php

namespace src\Behavioral\ChainOfResponsibility;

class RequestProcessor
{
    private MiddlewareChainHandler $middlewareChainHandler;

    /**
     * @param MiddlewareChainHandler $middlewareChainHandler
     */
    public function __construct(MiddlewareChainHandler $middlewareChainHandler)
    {
        $this->middlewareChainHandler = $middlewareChainHandler;
    }

    public function processRequest(Request $request)
    {
        if (!$this->middlewareChainHandler->handle($request)) {
            return new Response("Error processing request", false);
        }

        print_r("Processing request... \n");

        return new Response("Request Successfully Processed", true);
    }
}